src/main/java/com/hillel/items_exchange/service/LocationService.java

```diff
@@ -1,22 +1,70 @@
 package com.hillel.items_exchange.service;
 
-import com.hillel.items_exchange.dao.LocationRepository;
+import com.hillel.items_exchange.dto.LocationDto;
 import com.hillel.items_exchange.model.Location;
-import lombok.RequiredArgsConstructor;
-import org.springframework.stereotype.Service;
 
+import java.util.List;
 import java.util.Optional;
 
-@Service
-@RequiredArgsConstructor
-public class LocationService {
-    private final LocationRepository locationRepository;
+public interface LocationService {
+    /**
+     * Returns all Location DTOs.
+     * @return list of location entities from DB that are represented as {@link LocationDto}
+     */
+    List<LocationDto> findAll();
 
-    public Optional<Location> findById(long id) {
-        return locationRepository.findById(id);
-    }
+    /**
+     * Retrieves a location by its id.
+     * @param id Location ID.
+     * @return the location with the given id or Optional#empty() if none found.
+     */
+    Optional<Location> findById(long id);
 
-    public Location createLocation(Location location) {
-        return locationRepository.save(location);
-    }
-}
+    /**
+     * If a location exists, returns the Location DTO, otherwise throws {@code ClassNotFoundException}.
+     * @param id Location ID.
+     * @return the non-{@code null} {@link LocationDto} with given id.
+     * @throws ClassNotFoundException if no location found.
+     */
+    LocationDto getById(long id) throws ClassNotFoundException;
```

Comment:
Why don't you use Optional here?

Reply:
Because of "get" in the name of the method. I would like to name this method as "find", but I can't until `Optional<Location> findById(long id)` is removed. Also I can't use `protected` to overload the method because of  dealing with interface. Maybe you suggest me something clever at this case?